src/Join_SQL.php

```diff
@@ -119,9 +118,14 @@ public function initSelectQuery($model, $query)
                 isset($this->foreign_alias) ? (' '.$this->foreign_alias) : ''
             ),
             (
-                isset($this->owner->table_alias) ?
-                ($this->owner->table_alias.'.'.$this->master_field) :
-                ($this->owner->table).'.'.$this->master_field)
+                ($this->join
+                    // if join is nested, then use previous join table alias
+                    ? (isset($this->join->foreign_alias) ? $this->join->foreign_alias : $this->join->foreign_table)
+                    // otherwise use owner model table alias
+                    : (isset($this->owner->table_alias) ? $this->owner->table_alias : $this->owner->table)
+                ).'.'.$this->master_field
+            ),
+            $this->kind
         );
 
         if ($this->reverse) {
```

Comment:
Line next to this still looks suspicious. short_name = $this->join? really?
I have a feeling that something like I fixed few lines above should be there - with $this->join->foreign_alias/table if $this->join is set.
Please look into this before merging!